File: FinalProject/Contruction/ecobike_rental_server/TKXDPM_API/Model/Bike.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace TKXDPM_API.Model
{
    public class Bike
    {    
        [Key] public int BikeId { get; set; }
        [Column(TypeName = "varchar(255)")] public string BikeName { get; set; }
        [Column(TypeName = "varchar(512)")] public string BikeImage { get; set; }
        public string Description { get; set; }
        public int Size { get; set; }
        public int StartingRent { get; set; }
        public int HourlyRent { get; set; }
        public BikeType Type { get; set; }
        [Column(TypeName = "varchar(255)")] public string LicensePlates { get; set; }
        public int BatterCapacity { get; set; }
        public float PowerDrain { get; set; }
        public int Deposit { get; set; }

        public List<Rental> Rentals { get; set; }

        public Rental FindRental()
        {
            var rentals = Rentals.Where(rental => rental.Transaction.BookedEndDateTime == DateTime.MinValue).ToArray();
            return rentals[0];
        }
    }

    public class BikeResponse
    {
        public BikeResponse()
        {
            BikeId = 1;
            BikeName = "BikeName";
            BikeImage = "BikeImage";
            Description = "Address";
            StartingRent = 10;
            HourlyRent = 11;
            Type = BikeType.Double;
            LicensePlates = "Address";
            BatterCapacity = 10;
            PowerDrain = 0.6F;
            Deposit = 100000;
        }

        public int BikeId { get; set; }
        public string BikeName { get; set; }
        public string BikeImage { get; set; }
        public string Description { get; set; }
        public int StartingRent { get; set; }
        public int HourlyRent { get; set; }
        public BikeType Type { get; set; }
        public string LicensePlates { get; set; }
        public int BatterCapacity { get; set; }
        public float PowerDrain { get; set; }
        public int Deposit { get; set; }
    }

    public enum BikeType
    {
        Single = 1,
        Double = 2,
        Electric = 3,
    }
}

File: FinalProject/Contruction/ecobike_rental/lib/model/service/network/response/bike.dart
class BikeResponse {
  int bikeId;
  String bikeName;
  String description;
  int startingRent;
    String bikeImage;
  int hourlyRent;
  int type;
  String licensePlates;
  int batterCapacity;
  int powerDrain;
  int deposit;
  String rentals;

  BikeResponse(
      {this.bikeId,
      this.bikeName,
      this.description,
      this.startingRent,
         this.bikeImage,
      this.hourlyRent,
      this.type,
      this.licensePlates,
      this.batterCapacity,
      this.powerDrain,
      this.deposit,
      this.rentals});

  BikeResponse.fromJson(Map<String, dynamic> json) {
    bikeId = json['bikeId'];
    bikeName = json['bikeName'];
    description = json['description'];
    startingRent = json['startingRent'];
    hourlyRent = json['hourlyRent'];
    type = json['type'];
    bikeImage = json['bikeImage'];
    licensePlates = json['licensePlates'];
    batterCapacity = json['batterCapacity'];
    powerDrain = json['powerDrain'];
    deposit = json['deposit'];
    rentals = json['rentals'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['bikeId'] = this.bikeId;
    data['bikeName'] = this.bikeName;
    data['description'] = this.description;
    data['startingRent'] = this.startingRent;
    data['hourlyRent'] = this.hourlyRent;
    data['bikeImage'] = this.bikeImage;
    data['type'] = this.type;
    data['licensePlates'] = this.licensePlates;
    data['batterCapacity'] = this.batterCapacity;
    data['powerDrain'] = this.powerDrain;
    data['deposit'] = this.deposit;
    data['rentals'] = this.rentals;
    return data;
  }
}
